### DISCUSSION TITLE
tutorials/client-side-graphql-react/08-use-query-with-variables

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#94] [course feedback]
# tutorials/client-side-graphql-react/08-use-query-with-variables

https://www.apollographql.com/tutorials/client-side-graphql-react/08-use-query-with-variables

**Comment by traez** [comment]
Thanks so much for this. It's a great turorial! 
However this part won't work for people following your tutorial: "Now if we click a card, we go to that track's page."

Unless they check out the final branch on Github, and be sure to imitate, by adding "id" to
  const { title, thumbnail, author, length, modulesCount, id } = track;

  and to={`/track/${id}`} to
   <CardContainer to={`/track/${id}`}>

**Reply by mabuyo** [reply]
Hey there @traez, great catch, thanks so much for pointing this out! We'll get it fixed shortly :)

**Reply by traez** [reply]
You're welcome. This means so much to me. I'm a junior Dev, this is my first-week learning GraphQL.

**Comment by Victoremepunto** [comment]
Hi ! Maybe its just because I'm a React noob :) - but I was a bit puzzled due to the fact that the specific name of the variable used to extract the URL param actually mattered. 

I was trying to set the variable name to `trackID` , and while everything else from this example was updated accordingly (that is, my query's param also used `trackID` instead of this example's `trackId` the query would return a 404: NOT FOUND error. 

It turns out (again, this may not surprise someone with React knowledge) that the param is extracted from `src/pages/index.tsx` , where the `BrowserRouter` component defines the actual name your variable should use: 

```html
    <BrowserRouter>
      <Routes>
        <Route element={<Tracks />} path="/" />
        <Route element={<Track />} path="/track/:trackId" />   // <====
      </Routes>
    </BrowserRouter>
```

Sorry, I was puzzled - I did not connect the dots initially, this though taught me something about React :) - probably just a heads up for someone else hitting this (non) issue

**Reply by lizhennessy** [reply]
Hi there @Victoremepunto ! Thank you for sharing! That's definitely a tricky gotcha; we can imagine that the same `path` property could contain _multiple_ URL parameters, in which case it might be more apparent that the naming of each param matters! However, with just one, it's definitely a bit sneaky. :) Here's a link to an example using [URL parameters in the React Router documentation](https://v5.reactrouter.com/web/example/url-params) for anyone else who's curious about this feature!
